### PR TITLE
Added a CLI

### PR DESCRIPTION
**Pull request by quantum9Innovation**
# Basic CLI Structure Implemented
___

## New Feature
This PR adds a basic built-in CLI to epispot nightly that comes preinstalled with setup.py. The keyword to initiate it is `epispot` followed by a command. Currently the only available command is `about`. The shell uses `Fire` and `termcolor` as dependencies although it can run without `termcolor` by suppressing color (see the `__color__` flag). To test the shell, run:
```shell
$ epispot about
>>> Epispot was invoked via the epispot CLI
>>> Path: /home/user/anaconda3/lib/python3.8/site-packages/epispot_nightly-2.1.1.1-py3.8.egg/EGG-INFO/scripts
>>> Version: shell-v0.1.0-alpha epispot-v2.1.1.1
>>> Color output enabled (True/False): True

```
## Known Issues
- Coverage does not work on CLI scripts.
- Planning on adding an enhancement to run models from the CLI; will require the addition of automatic 
parameter substitutions that will be implemented in another branch.
## Code Breakdown

 - .coveragerc
    - Added bin/ as a directory however still not implemented in codecov reports (issue)
 - .github/workflows
    - Added to support build tests for CLI
    - coverage.yml
        - Added more verbosity for issue solving related to code coverage reporting
- CHANGELOG.md
    - Added a CHANGELOG to track progress on the nightly package
- README.md
    - Removed Release Notes on README.md to create CHANGELOG.md
- bin/requirements.txt
    - Added separate CLI requirements (Fire, termcolor (optional))
- requirements.txt
    - Removed unnecessary requirements (sphinx)
- setup.py
    - Added `scripts` indicator to add `epispot` to the command line
## Additional Notes
Similar to branch #cli
Ideally merge when:
 - Code and implementation has been thoroughly reviewed
 - New workflows have been approved by @Quantalabs 's review
**Ignoring:**
 - LGTM, CodeFactor, and other code quality checks (except built-in GitHub CodeQL)
     - Currently transitioning to Deepsource

### CODE REVIEW

**Comment by quantum9Innovation** [comment]
Ready to merge after workflow approvals from @Quantalabs

File: .github/workflows/build.yml
Comment (Quantalabs) [inline .github/workflows/build.yml:45]: Instead of installing, it's better to use `develop`.

```suggestion
        python setup.py develop
```

File: bin/epispot
Comment (Quantalabs) [inline bin/epispot:44]: Why are you using a library like termcolor for terminal string styling? You can use the python built-in ANSI escape codes which are much more customizable, such as _italics_ or underlined

File: setup.py
Comment (Quantalabs) [inline setup.py:15]: You could also use `console_scripts` which will allow for more customizable command names.

```suggestion
    console_scripts=[
         'epi: bin/epispot'
    ]
```

**Review by Quantalabs — CHANGES_REQUESTED** [review]

File: setup.py
Comment (Quantalabs) [reply setup.py:15]: You should check though that this works with classes.

Comment (Quantalabs) [reply setup.py:15]: It seems this doesn't work with classes. Ignore this suggestion.

**Comment by quantum9Innovation** [comment]
Using #40 instead; keeping this PR as a backup in case anything goes wrong.

**Comment by Quantalabs** [comment]
Will re-open if any bad problems turn up in #40

**Pull request merged**
